firefox-ios/Client/TabManagement/Tab.swift

```diff
@@ -693,7 +693,8 @@ class Tab: NSObject, ThemeApplicable, FeatureFlaggable, ShareTab {
     func reload(bypassCache: Bool = false) {
         // If the current page is an error page, and the reload button is tapped, load the original URL
         if let url = webView?.url, let internalUrl = InternalURL(url), let page = internalUrl.originalURLFromErrorPage {
-            webView?.replaceLocation(with: page)
+            let request = URLRequest(url: page, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData)
+            webView?.load(request)
```

Comment:
I guess this change is needed since our error page are now native instead of being html error pages? I am wondering if it's worth investigating in another task if `replaceLocation(with:)` can be entirely removed (and this reload function could also be simplified). Replace location is used in two other places in the project, where one relates to error page as well, and the other one relates to session restore which doesn't exist anymore (so it's probably dead code). What do you think? I can create the task for this!

Reply:
A follow-up ticket will be perfect. Please create it and you could assign me to it.

Reply:
Here is the ticket https://mozilla-hub.atlassian.net/browse/FXIOS-10883. I looked a bit and I think we'll need to wait for this clean up thought, since not all native error pages exists yet.

Reply:
That brings me to the next question, for this PR, I think we should check if `featureFlags.isFeatureEnabled(.nativeErrorPage, checking: .buildOnly)` is enabled?

Reply:
To clarify, this PR is not related to `nativeErrorPage`. It addresses a bug that causes the previously cached website to load upon refreshing, even without an internet connection. This issue occurs regardless of whether the flag is enabled. I modified the webview to load a new request, ignoring the cache, to accurately reflect the current state of the page, which in our case is an error.

Reply:
Ok thanks for explaining! I thought it related 🙏